src/__tests__/pyunicoreWidget.test.tsx

```diff
@@ -0,0 +1,132 @@
+// mock the handler for api requests
+jest.mock('../handler', () => {
+  return {
+    __esModule: true,
+    requestAPI: jest
+      .fn()
+      .mockImplementation((url, init, settings) => Promise.resolve(data))
+  };
+});
+// mock the ReactWidget
+jest.mock('@jupyterlab/apputils', () => {
+  return {
+    __esModule: true,
+    ReactWidget: jest.fn().mockImplementation(() => {
+      return {
+        render: () => {
+          console.log('render');
+        }
+      };
+    })
+  };
+});
+
+import { IJob, PyunicoreComponent } from '../pyunicoreWidget';
+import {
+  findByText,
+  fireEvent,
+  getByTestId,
+  render,
+  waitFor
+} from '@testing-library/react';
+
+import {
+  BUTTON_SETTINGS,
+  getKernel,
+  mockGetJob
+} from './UnicoreJobsTable.test';
+import React from 'react';
+
+const JUDAC = 'JUDAC';
+const JUSUF = 'JUSUF';
+const data = {
+  message: '',
+  jobs: generateJobs(12)
+};
+
+export function generateJobs(count: number): Array<IJob> {
+  const generatedJobs = [];
+  // add more jobs to have next button visible
+  for (let i = 0; i < count; i++) {
+    generatedJobs.push({
+      id: `test${i}`,
+      name: `test_name_${i}`,
+      owner: 'test',
+      site: JUDAC,
+      status: 'FAILED',
+      resource_url: `test_url_2${i}`,
+      start_time: '2022-02-18T10:54:08+0100',
+      is_cancelable: false,
+      logs: ['line 1', 'line 2']
+    });
+  }
+  return generatedJobs;
+}
+
+const RELOAD_RATE_MS = 60000;
```

Comment:
Regarding this, I meant this constant could be used by both react component and the tests, instead of keeping this value hardcoded in both places. Thus, in case we want to change its value in the future, we will do it in a single place.